scripts/main.js

```diff
@@ -1,32 +1,37 @@
 $("#new-app-btn").on("click", function(){
-$("#apps-container").hide();
-$("#create-app-container").show();
-$("#detail-app-container").hide();
-	//create an new app, when a user click 'create button'
-var createBtn = document.querySelector("#choose-create");
-if(createBtn){
-  createBtn.addEventListener("click",function(event){
-			//create app array for exisiting apps
-	var appname = document.getElementsByTagName("input")[0].value;
-	var app = { name: appname};
-	if(app.name){
-		applist.add ({
-		  name : appname
-		})
-	makeAppList();
-	showAppDetail(app.id);
-					    }
-		      });
-	    }
+  $("#apps-container").hide();
+  $("#create-app-container").show();
+  $("#detail-app-container").hide();
+  	//create an new app, when a user click 'create button'
+  var createBtn = document.querySelector("#choose-create");
+  if(createBtn){
+    createBtn.addEventListener("click",function(event){
+    	//create app array for exisiting apps
+    	var appname = document.getElementsByTagName("input")[0].value;
+    	var app = { name: appname};
+    	if(app.name){
+    		applist.add ({
+    		  name : appname
+    		})
+        .then(function (app) {
+          makeAppList();
+        	showAppDetail(app.id);
+        })
+      }
+  });
+}
 });
 function showAppDetail(id) {
-  var app = applist.find(id);
```

Comment:
similar problem as above. `app` here is a promise, because `applist.find()` returns a promise, not the actual app object. I know that is a bit complex, but unfortunately we have to make these APIs asynchronous as we will write to files and/or databases in future, and we cannot make this a synchronous process that blocks the execution of other JavaScript. This is why we use Promises